### PR TITLE
[Helix] Templates tests 2.0

### PR DESCRIPTION
**Pull request by HaoK**
Figure out restore issues

### CODE REVIEW

**Comment by HaoK** [comment]
@dougbu do you know where EntityFramework 5.0.0-ci comes from during the normal template builds/test runs?  I'm guessing I just need to include it into the helix payload for the templates tests

**Comment by dougbu** [comment]
@ryanbrandenburg I am not sure of the answer to @HaoK's question. We have pinned most efcore dependencies in this repo but I'm guessing the templates aren't aware of that? How would we use the same (pinned) versions in the templates? (Help plz)

**Comment by HaoK** [comment]
Looks like adding to the nuget sources directly via dotnet nuget add configures the feeds properly on the helix machines, so no need to go through properties anymore.  So I just need to a way to get the nuget packages we aren't building, namely it looks like, extensions, EF, and razor?

Microsoft.EntityFrameworkCore.Relational with version (>= 5.0.0-preview.2.20126.1)
Microsoft.EntityFrameworkCore.SqlServer with version (>= 5.0.0-preview.2.20126.1)
Microsoft.EntityFrameworkCore.Tools with version (>= 5.0.0-preview.2.20126.1)
Microsoft.Extensions.DependencyModel with version (>= 5.0.0-preview.2.20125.16)
Microsoft.CodeAnalysis.Razor with version (>= 5.0.0-preview.2.20126.1)
Microsoft.Extensions.Logging with version (>= 5.0.0-preview.2.20126.1)
Microsoft.Extensions.Caching.Abstractions with version (>= 5.0.0-preview.2.20126.1)
Microsoft.Extensions.FileProviders.Abstractions with version (>= 5.0.0-preview.2.20126.1)

**Comment by HaoK** [comment]
Nice @dougbu that feed works perfectly, fixing the ef path, and that should result in 52/65 test passing, the 13 skipped tests are selenium tests

**Comment by HaoK** [comment]
Reenables tests that were disabled due to package restore and dot net ef not working

File: eng/helix/content/runtests.sh [hunk 1]
```diff
@@ -95,6 +98,17 @@ if [ -d "Microsoft.AspNetCore.App" ]
...
+    
+    echo "Adding current directory to nuget sources: $DIR"
+    dotnet nuget add source $DIR
+    dotnet nuget add source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json
+    dotnet nuget list source
+    
+    dotnet tool install dotnet-ef --global --version $efVersion
+    
+    # Ensure tools are on on PATH
+    export PATH="$PATH:$DOTNET_CLI_HOME/.dotnet/tools"
```
Comment (dougbu) [inline eng/helix/content/runtests.sh]: Why does this directory need to be on the path? Won't `dotnet` find it automatically?

File: src/ProjectTemplates/test/IdentityUIPackageTest.cs [hunk 2]
```diff
@@ -118,9 +118,8 @@ public IdentityUIPackageTest(ProjectFactoryFixture projectFactory, ITestOutputHe
             "Identity/lib/jquery-validation-unobtrusive/LICENSE.txt",
         };
 
-        [ConditionalTheory]
+        [Theory]
```
Comment (dougbu) [inline src/ProjectTemplates/test/IdentityUIPackageTest.cs]: This was the only conditional test. Should be able to remove
``` c#
using Microsoft.AspNetCore.Testing;
```

File: src/ProjectTemplates/test/RazorClassLibraryTemplateTest.cs [hunk 3]
```diff
@@ -41,8 +41,7 @@ public async Task RazorClassLibraryTemplate_WithViews_Async()
             Assert.True(0 == buildResult.ExitCode, ErrorMessages.GetFailedProcessMessage("build", Project, buildResult));
         }
 
-        [ConditionalFact]
-        [SkipOnHelix("restore no worky")]
+        [Fact]
```
Comment (dougbu) [inline src/ProjectTemplates/test/RazorClassLibraryTemplateTest.cs]: Can remove and sort excess `using`s here too.

**Review by dougbu — CHANGES_REQUESTED** [review]
Please remove and sort `using`s in all files. Suspect there's only a couple of excess `using`s but the don't seem to be sorted in any of these files.

File: eng/helix/content/runtests.sh (hunk 1 above)
Comment (HaoK) [reply eng/helix/content/runtests.sh]: Nope, probably because we install the CLI in this script

**Comment by HaoK** [comment]
Look good now @dougbu ?

**Review by dougbu — APPROVED** [review]
The `using`s don't match our guidelines in a number of these files because they're not sorted but I'm not going to block this PR on it.

I think the main issue is where you added `Microsoft.AspNetCore.Testing` recently. Please file a follow-up issue or added this to the existing one.

**Comment by HaoK** [comment]
Sure I'll fix that in the next iteration

**Pull request merged**